spec/models/provider_spec.rb

```diff
@@ -61,4 +61,23 @@
       expect(tenant.providers).to include(provider)
     end
   end
+
+  context "#destroy_queue" do
+    it "destroy when has no managers" do
+      expect(provider).to receive(:destroy)
+      provider.destroy_queue
+    end
+
+    it "to destroy_queue its managers and itself" do
+      manager = FactoryGirl.create(:ext_management_system, :zone => EvmSpecHelper.local_miq_server.zone)
```

Comment:
I thought an EMS factory uses the local zone by default?

Reply:
It's not, I have to add this `:zone` in order not to fail at `nil` zone